### DISCUSSION TITLE
issues with make Darknet

### DISCUSSION THREAD

**Discussion by romz01** [AlexeyAB/darknet#8741] [Q&A]
Hello everybody,

I have a problem when i try to compilate darknet.

I was using darknet  in conda environment.
It had been a while since I had used this environment and when I reused it there were error messages when trying to start darknet again
I decided to try recompiling Darknet, I thought there must have been updates and that recompiling would solve my problems.
I even uninstalled and reinstalled conda to be sure.
But the problem is still the same I have this error message when I run the make:

nvcc -gencode arch=compute_86,code=[sm_86,compute_86] -Iinclude/ -I3rdparty/stb/include -I/home/romain/anaconda3/envs/YOLOv4_conda/include -DOPENCV `pkg-config --cflags opencv4 2> /dev/null || pkg-config --cflags opencv` -DGPU -I/usr/local/cuda/include/ -DCUDNN --compiler-options "-Wall -Wfatal-errors -Wno-unused-result -Wno-unknown-pragmas -fPIC -Ofast -DOPENCV -DGPU -DCUDNN -I/usr/local/cudnn/include" -c ./src/avgpool_layer_kernels.cu -o obj/avgpool_layer_kernels.o
g++ -std=c++11 -std=c++11 -Iinclude/ -I3rdparty/stb/include -I/home/romain/anaconda3/envs/YOLOv4_conda/include -DOPENCV `pkg-config --cflags opencv4 2> /dev/null || pkg-config --cflags opencv` -DGPU -I/usr/local/cuda/include/ -DCUDNN -Wall -Wfatal-errors -Wno-unused-result -Wno-unknown-pragmas -fPIC -Ofast -DOPENCV -DGPU -DCUDNN -I/usr/local/cudnn/include obj/image_opencv.o obj/http_stream.o obj/gemm.o obj/utils.o obj/dark_cuda.o obj/convolutional_layer.o obj/list.o obj/image.o obj/activations.o obj/im2col.o obj/col2im.o obj/blas.o obj/crop_layer.o obj/dropout_layer.o obj/maxpool_layer.o obj/softmax_layer.o obj/data.o obj/matrix.o obj/network.o obj/connected_layer.o obj/cost_layer.o obj/parser.o obj/option_list.o obj/darknet.o obj/detection_layer.o obj/captcha.o obj/route_layer.o obj/writing.o obj/box.o obj/nightmare.o obj/normalization_layer.o obj/avgpool_layer.o obj/coco.o obj/dice.o obj/yolo.o obj/detector.o obj/layer.o obj/compare.o obj/classifier.o obj/local_layer.o obj/swag.o obj/shortcut_layer.o obj/representation_layer.o obj/activation_layer.o obj/rnn_layer.o obj/gru_layer.o obj/rnn.o obj/rnn_vid.o obj/crnn_layer.o obj/demo.o obj/tag.o obj/cifar.o obj/go.o obj/batchnorm_layer.o obj/art.o obj/region_layer.o obj/reorg_layer.o obj/reorg_old_layer.o obj/super.o obj/voxel.o obj/tree.o obj/yolo_layer.o obj/gaussian_yolo_layer.o obj/upsample_layer.o obj/lstm_layer.o obj/conv_lstm_layer.o obj/scale_channels_layer.o obj/sam_layer.o obj/convolutional_kernels.o obj/activation_kernels.o obj/im2col_kernels.o obj/col2im_kernels.o obj/blas_kernels.o obj/crop_layer_kernels.o obj/dropout_layer_kernels.o obj/maxpool_layer_kernels.o obj/network_kernels.o obj/avgpool_layer_kernels.o -o darknet -lm -pthread `pkg-config --libs opencv4 2> /dev/null || pkg-config --libs opencv` -L/usr/local/cuda/lib64 -lcuda -lcudart -lcublas -lcurand -L/usr/local/cudnn/lib64 -lcudnn -lstdc++
/usr/bin/ld : ne peut pas trouver -lcudart : Aucun fichier ou dossier de ce type
/usr/bin/ld : ne peut pas trouver -lcublas : Aucun fichier ou dossier de ce type
/usr/bin/ld : ne peut pas trouver -lcurand : Aucun fichier ou dossier de ce type
/usr/bin/ld : ne peut pas trouver -lcudnn : Aucun fichier ou dossier de ce type
collect2: error: ld returned 1 exit status
make: *** [Makefile:176 : darknet] Erreur 1

Anyone have an idea to help me?

**Comment by stephanecharette** [comment]
You don't have CUDA and CUDNN installed:

> /usr/bin/ld : ne peut pas trouver -lcudart : Aucun fichier ou dossier de ce type
> /usr/bin/ld : ne peut pas trouver -lcublas : Aucun fichier ou dossier de ce type
> /usr/bin/ld : ne peut pas trouver -lcurand : Aucun fichier ou dossier de ce type
> /usr/bin/ld : ne peut pas trouver -lcudnn : Aucun fichier ou dossier de ce type

**Reply by romz01** [reply]
thank's for answer but i think CUDA and CUDNN are installed  : 
![Capture d’écran du 2023-02-20 08-42-12](https://user-images.githubusercontent.com/102368428/220043211-370918ec-a270-41ab-acf8-b3bdfcd772f2.png)


That's why i don't understand why darknet is not compilated ..

**Reply by romz01** [reply]
But i try to compile darkent in a conda environnement and i think that make precssus try to find Cuda and Cudnn in global envornnement because of it search in : "/usr/bin/ld". 
I have wright ? 

I change directory in the makefile but it doesn't change anything ...
